pipeval/validate/files.py

```diff
@@ -1,16 +1,52 @@
 ''' File checking functions '''
 from pathlib import Path
+from typing import Union
 import warnings
+import zlib
+import gzip
+import bz2
 import magic
 
-def _check_compressed(path:Path):
-    ''' Check file is compressed '''
-    compression_mimes = [
-        'application/x-gzip',
-        'application/x-bzip2'
-    ]
-    if magic.from_file(path.resolve(), mime=True) not in compression_mimes:
+def _identify_compression(path:Path):
+    ''' Identify compression type and returns appropriate file handler '''
+    compression_handlers = {
+        'application/x-gzip': gzip.open,
+        'application/x-bzip2': bz2.open
+    }
+
+    file_mime = magic.from_file(path.resolve(), mime=True)
+    return compression_handlers.get(file_mime, None)
+
+def _check_compression_integrity(path:Path, handler:Union[gzip.open,bz2.open]):
+    ''' Verify integrity of compressed file '''
+    integrity_error = ''
+    read_chunk_size = 1000000 # 1 MB chunks
+
+    with handler(path, 'rb') as file_reader:
+        try:
+            while file_reader.read(read_chunk_size) != b'':
+                pass
+        except gzip.BadGzipFile as bad_gzip:
+            integrity_error = f'Invalid Gzip file: {str(bad_gzip)}'
```

Comment:
I don't believe calling `str` is necessary with string interpolation. Python should do this automatically.

Reply:
You are correct, I've removed `str`